src/sst.rs

```diff
@@ -18,26 +18,60 @@ use std::io::Read;
 #[cfg(feature = "lz4")]
 use std::io::Write;
 
+pub(crate) struct SsTableFormatBuilder {
+    format: SsTableFormat,
+}
+
+impl SsTableFormatBuilder {
+    pub(crate) fn new() -> Self {
+        Self {
+            format: SsTableFormat {
+                block_size: 4096,
+                min_filter_keys: 0,
+                filter_bits_per_key: 10,
+                compression_codec: None,
+            },
+        }
+    }
+
+    #[allow(dead_code)]
+    pub(crate) fn with_block_size(mut self, block_size: usize) -> Self {
+        self.format.block_size = block_size;
+        self
+    }
+
+    pub(crate) fn with_min_filter_keys(mut self, min_filter_keys: u32) -> Self {
+        self.format.min_filter_keys = min_filter_keys;
+        self
+    }
+
+    pub(crate) fn with_filter_bits_per_key(mut self, filter_bits_per_key: u32) -> Self {
+        self.format.filter_bits_per_key = filter_bits_per_key;
+        self
+    }
+
+    pub(crate) fn with_compression_codec(
+        mut self,
+        compression_codec: Option<CompressionCodec>,
+    ) -> Self {
+        self.format.compression_codec = compression_codec;
+        self
+    }
+
+    pub(crate) fn build(self) -> SsTableFormat {
+        self.format
+    }
+}
+
 #[derive(Clone)]
 pub(crate) struct SsTableFormat {
     block_size: usize,
     min_filter_keys: u32,
+    filter_bits_per_key: u32,
     compression_codec: Option<CompressionCodec>,
 }
 
 impl SsTableFormat {
-    pub fn new(
```

Comment:
Even if we go with the `Default::default()` style that I described above, I guess we still don't need `::new` since we can just do `SsTableFormat { ... }` to create the struct directly.